### PR TITLE
fixed bugs #36972 and #34657

### PR DESCRIPTION
**Pull request by c-knof**

### CODE REVIEW

File: Services/Language/classes/Setup/class.ilSetupLanguage.php [hunk 1]
```diff
@@ -584,13 +584,28 @@ protected function insertLanguage($lang_key, $scope = '')
...
+                $modules_to_delete = "module = '";
+                $count_modules = count($lang_array);
+                foreach ($lang_array as $module => $lang_arr) {
+                    $modules_to_delete .= $module . "'";
+                    $count_modules -= 1;
+                    if ($count_modules) {
+                        $modules_to_delete .= " OR module = '";
+                    }
+                }
+                $delete_query .= " AND (" . $modules_to_delete . ")";
```
Comment (mjansenDatabay) [inline Services/Language/classes/Setup/class.ilSetupLanguage.php]: Please use `ilDBInterface::quote` and don't write values unescaped into the query.

File: Services/Language/classes/class.ilObjLanguageDBAccess.php [hunk 2]
```diff
@@ -134,13 +134,29 @@ public function replaceLangModules(array $lang_array): void
...
+            $modules_to_delete = "module = '";
+            $count_modules = count($lang_array);
+            foreach ($lang_array as $module => $lang_arr) {
+                $modules_to_delete .= $module . "'";
+                $count_modules -= 1;
+                if ($count_modules) {
+                    $modules_to_delete .= " OR module = '";
+                }
+            }
+            $delete_query .= " AND (" . $modules_to_delete . ")";
```
Comment (mjansenDatabay) [inline Services/Language/classes/class.ilObjLanguageDBAccess.php]: Please use `ilDBInterface::quote` and don't write values unescaped into the query.

File: Services/Language/classes/Setup/class.ilSetupLanguage.php (hunk 1 above)
Comment (c-knof) [reply Services/Language/classes/Setup/class.ilSetupLanguage.php]: done

File: Services/Language/classes/class.ilObjLanguageDBAccess.php (hunk 2 above)
Comment (c-knof) [reply Services/Language/classes/class.ilObjLanguageDBAccess.php]: done

**Review by mjansenDatabay — APPROVED** [review]

**Pull request merged**